epochStart/metachain/systemSCs.go

```diff
@@ -471,6 +484,99 @@ func (s *systemSCProcessor) setMinNumberOfNodes(minNumNodes uint32) error {
 	return nil
 }
 
+func (s *systemSCProcessor) updateOwnersForBlsKeys() error {
+	sw := core.NewStopWatch()
+	sw.Start("systemSCProcessor")
+	defer func() {
+		sw.Stop("systemSCProcessor")
+		log.Debug("systemSCProcessor.updateOwnersForBlsKeys time measurements", sw.GetMeasurements())
+	}()
+
+	userAuctionAccount, err := s.getAuctionSystemAccount()
+	if err != nil {
+		return err
+	}
+
+	auctionAccounts, err := s.getValidAuctionUserAccountsKeys(userAuctionAccount)
+	if err != nil {
+		return err
+	}
+
+	err = s.callUpdateStakingV2(auctionAccounts)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *systemSCProcessor) getAuctionSystemAccount() (state.UserAccountHandler, error) {
+	auctionAccount, err := s.userAccountsDB.LoadAccount(vm.AuctionSCAddress)
+	if err != nil {
+		return nil, fmt.Errorf("%w when loading auction account", err)
+	}
+
+	userAuctionAccount, ok := auctionAccount.(state.UserAccountHandler)
+	if !ok {
+		return nil, fmt.Errorf("%w when loading auction account", epochStart.ErrWrongTypeAssertion)
+	}
+
+	if check.IfNil(userAuctionAccount.DataTrie()) {
+		return nil, epochStart.ErrNilDataTrie
+	}
+
+	return userAuctionAccount, nil
+}
+
+func (s *systemSCProcessor) getValidAuctionUserAccountsKeys(userAuctionAccount state.UserAccountHandler) ([][]byte, error) {
+	auctionAccounts := make([][]byte, 0)
+	chLeaves := userAuctionAccount.DataTrie().GetAllLeavesOnChannel()
+	for leaf := range chLeaves {
+		auctionData := &systemSmartContracts.AuctionDataV2{}
```

Comment:
I think it would be enough to check the length of the keys that they are addresses :). but this happens only in one single epoch so it can stay.